src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyAndFieldNamesTests.Fixer.cs

```diff
@@ -0,0 +1,384 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.ConstructorParametersShouldMatchPropertyAndFieldNamesAnalyzer,
+    Microsoft.NetCore.Analyzers.Runtime.ConstructorParametersShouldMatchPropertyAndFieldNamesFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.ConstructorParametersShouldMatchPropertyAndFieldNamesAnalyzer,
+    Microsoft.NetCore.Analyzers.Runtime.ConstructorParametersShouldMatchPropertyAndFieldNamesFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class ConstructorParametersShouldMatchPropertyAndFieldNamesFixerTests
```

Comment:
⚠️ ⚠️ There shouldn't be separate test files for analyzer and codefix.

Reply:
Why so? It seems like a lot of analyzers have separate files for analyzer and a code fix. For example:
- TestForEmptyStringsUsingStringLengthTests
  TestForEmptyStringsUsingStringLengthTests.Fixer
- UseOrdinalStringComparisonTests 
   UseOrdinalStringComparisonTests.Fixer
- MarkAllNonSerializableFieldsTests
  MarkAllNonSerializableFieldsTests.Fixer
etc.

Reply:
For me it's OK to have separate files, especially for the scenarios where codefix is not suggested, testing no diagnostic /false positives scenarios. 

Though for the scenarios that has a codefix do not need to repeat the test in the analyzers test and codefix test. For example I see you have a same test in analyzer and code fix:

````cs
        [Fact]
        public async Task CA1071_ClassPropsDoNotMatch_ConstructorParametersShouldMatchPropertyNames_CSharp()
        {
             await VerifyCSharpAnalyzerAsync(@"
                using System.Text.Json.Serialization;
                public class C1
                {
                    public int FirstProp { get; }
                    public object SecondProp { get; }
                    [JsonConstructor]
                    public C1(int {|#0:firstDrop|}, object {|#1:secondDrop|})
                    {
                        this.FirstProp = firstDrop;
                        this.SecondProp = secondDrop;
                    }
                }",
                CA1071CSharpPropertyOrFieldResultAt(0, "C1", "firstDrop", "FirstProp"),
                CA1071CSharpPropertyOrFieldResultAt(1, "C1", "secondDrop", "SecondProp"));
         }
```
And in code fix tests:
```cs
        [Fact]
        public async Task CA1071_ClassSinglePropDoesNotMatch_CSharp()
        {
            await VerifyCSharpCodeFixAsync(@"
                using System.Text.Json.Serialization;
                public class C1
                {
                    public int FirstProp { get; }
                    public object SecondProp { get; }
                    [JsonConstructor]
                    public C1(int [|firstDrop|], object secondProp)
                    {
                        this.FirstProp = firstDrop;
                        this.SecondProp = secondProp;
                    }
                }",
                @"
                using System.Text.Json.Serialization;
                public class C1
                {
                    public int FirstProp { get; }
                    public object SecondProp { get; }
                    [JsonConstructor]
                    public C1(int firstProp, object secondProp)
                    {
                        this.FirstProp = firstProp;
                        this.SecondProp = secondProp;
                    }
                }");
        }
```
These should be just in code fix tests